### PR TITLE
[KINDA MODULAR] Claws Cuirk

### PR DESCRIPTION
**Pull request by Nerev4r**
## About The Pull Request
Remember Warwick from [League of Legends](https://en.wikipedia.org/wiki/League_of_Legends), how he had claws built into his fingers?
image
These aren't super cool steampunk claws, it just makes your hands do a claw effect and sharp damage. No increase in stamina or wound bonus or anything.

## How This Contributes To The Skyrat Roleplay Experience
image
So this isn't actually that crazy. The wound bonus on unarmed damage is pretty low, I tested with a kitchen knife (the easiest source of SHARP_EDGED) and it was **far** easier to go all Jack the Ripper on people than this quirk; fist-to-fist against a monkeybrained human, you'd still not have a _massive_ advantage in unarmed unless you get pretty lucky, and bleeding is easier to fix than most blunt wounds. I simply think this will be neat for character creation and to further accentuate certain kinds of character.

Also it's kinda hot ngl.

## Proof of Testing
<details>
<summary>Screenshots/Videos</summary>
  
image

</details>

## Changelog
:cl:
add: NanoTrasen grooming regulations no longer specify maximum fingernail length, leading to more clawed employees appearing on nearby facilities.
/:cl:

### CODE REVIEW

**Comment by Plum-but-gayer** [comment]
love it - great step towards more fun roundstart flavor stuff that effects mechanics - but also would prefer the razor girl name to be saved for when we add the actual retractable blade from sojourn, something that is real and will definitely happen.

**Comment by Nerev4r** [comment]
image
Good point.

**Comment by kittysmooch** [comment]
mandatory for felinds. almost seems kind of expensive for a quirk that's not super good at all but then again spacer costs 7 and is borderline useless outside of spacetide.

**Comment by PCPort** [comment]
New barber meta.
edward-scissorhands-johnny-depp

File: code/modules/surgery/bodyparts/_bodyparts.dm [hunk 1]
```diff
@@ -192,7 +192,7 @@
 	var/unarmed_stun_threshold = 2
 	/// How many pixels this bodypart will offset the top half of the mob, used for abnormally sized torsos and legs
 	var/top_offset = 0
-
+	var/unarmed_sharpness = NONE
```
Comment (softcerv) [inline code/modules/surgery/bodyparts/_bodyparts.dm]: You could put an autodoc comment on this and also move this to a modular file

File: code/modules/mob/living/carbon/human/_species.dm [hunk 2]
```diff
@@ -1208,14 +1208,15 @@ GLOBAL_LIST_EMPTY(features_by_species)
...
+		var/unarmed_sharpness = attacking_bodypart.unarmed_sharpness //SKYRAT EDIT - If unarmed damage sharpness needs to be taken into account.
 		if(atk_effect == ATTACK_EFFECT_KICK)//kicks deal 1.5x raw damage
 			if(damage >= 9)
 				target.force_say()
 			log_combat(user, target, "kicked")
 			target.apply_damage(damage * PUNCH_STAMINA_MULTIPLIER, STAMINA, affecting, armor_block) //SKYRAT EDIT ADDITION
 			target.apply_damage(damage, attack_type, affecting, armor_block, attack_direction = attack_direction)
 		else//other attacks deal full raw damage + 1.5x in stamina damage
-			target.apply_damage(damage, attack_type, affecting, armor_block, attack_direction = attack_direction)
+			target.apply_damage(damage, attack_type, affecting, armor_block, attack_direction = attack_direction, sharpness = unarmed_sharpness) //SKYRAT EDIT CHANGE: Applies sharpness if it does.
```
Comment (softcerv) [inline code/modules/mob/living/carbon/human/_species.dm]: You should keep the original line this replaces commented out but still in the code

File: modular_skyrat/master_files/code/datums/traits/good.dm [hunk 3]
```diff
@@ -20,6 +20,51 @@
...
+	name = "Sharp Claws"
+	desc = "Whether it's a hunter's inherent biology, or your stubborn refusal to clip your nails ahead of your Jiu-Jitsu classes, your unarmed attacks are sharper and might make folks bleed."
+	value = 6 // It's still not really as strong as a kitchen knife in terms of wounding, but we can go to eight if we gotta.
+	gain_text = span_notice("Your palms hurt a bit from the sharpness of your nails.")
+	lose_text = span_danger("You feel a distinct emptiness as your nails dull; good luck scratching that itch.")
+	medical_record_text = "Patient ended up scratching through the examination table's cushions; recommended they look into clipping their claws."
+	icon = FA_ICON_HAND
+
+/datum/quirk/sharpclaws/add(client/client_source)
+	var/mob/living/carbon/human/human_holder = quirk_holder
```
Comment (softcerv) [inline modular_skyrat/master_files/code/datums/traits/good.dm]:
```suggestion
	var/mob/living/carbon/human/human_holder = quirk_holder
	if(!istype(human_holder))
	        return FALSE
	        
```
better safe than sorry

File: code/modules/surgery/bodyparts/_bodyparts.dm (hunk 1 above)
Comment (softcerv) [inline code/modules/surgery/bodyparts/_bodyparts.dm]:
```suggestion
	var/unarmed_sharpness = 0
```

**Comment by HoodMilk** [comment]
Really expensive but nice and flavorful. Just make it 2 cost or neutral, tier 1 bleeding wounds don't cause enough blood loss to matter and heal themselves quickly.  People concerned with balance should be reminded that every emergency locker in the game spawns with deadlier weapons, and most jobs spawn with more robust tools than fists.

**Comment by Lafrien** [comment]
May i ask what happens if you insert a Strong-Arm implant? Will the sharpness gets nullified, or the combo might become rather deadly?

**Comment by Nerev4r** [comment]
image
Lowered the cost.

**Comment by YakumoChen** [comment]
Is this effect blocked by gloves?
Is it blocked by gloves that specifically change your martial arts (most notably boxing gloves)?
Other martial arts?
Chunky fingers? (Hulk and insulated especially)

**Comment by Nerev4r** [comment]
Okay, so.
Strong Arm implants use their own damage proc, so this doesn't impact that.
Martial arts use their own damage proc and so does boxing/krav maga, so it doesn't end up sharp.
Hulk has its' own proc too.
I probably won't make thick gloves block it because anthros/lizards too (even tho it's cosmetic) can scratch through them and most importantly, Ethereals can apply unarmed burn damage through them.

**Comment by Nerev4r** [comment]
fuck

**Comment by Nerev4r** [comment]
help lmao

**Comment by Nerev4r** [comment]
head on over to #22869

**Pull request closed without merging**